nmigen/build/generic_platform.py

```diff
@@ -0,0 +1,161 @@
+from collections import defaultdict, OrderedDict
```

Comment:
This file (and its corresponding test file) should really be called something like `nmigen.build.constraint`.